scripts/blueprints/dockerhub/node.md

```diff
@@ -0,0 +1,148 @@
+
+> [wikipedia.org/wiki/Node.js](https://en.wikipedia.org/wiki/Node.js)
+
+![logo](https://raw.githubusercontent.com/docker-library/docs/01c12653951b2fe592c1f93a13b4e289ada0e3a1/node/logo.png)
+
+# How to use this image
+
+### Create a `Dockerfile` in your Node.js app project
+
+```dockerfile
+# specify the node base image with your desired version node:<version>
+FROM balenalib/raspberrypi3-node:10
+# replace this with your application's default port
+EXPOSE 8888
+```
+
+You can then build and run the Docker image:
+
+```console
+$ docker build -t my-nodejs-app .
+$ docker run -it --rm --name my-running-app my-nodejs-app
+```
+
+If you prefer Docker Compose:
+
+```yml
+version: "2"
+services:
+  node:
+    image: "balenalib/raspberrypi3-node:10"
+    user: "node"
+    working_dir: /home/node/app
+    environment:
+      - NODE_ENV=production
+    volumes:
+      - ./:/home/node/app
+    expose:
+      - "8081"
+    command: "npm start"
+```
+
+You can then run using Docker Compose:
+
+```console
+$ docker-compose up -d
+```
+
+Docker Compose example copies your current directory (including node_modules) to the container.
+It assumes that your application has a file named [`package.json`](https://docs.npmjs.com/files/package.json)
+defining [start script](https://docs.npmjs.com/misc/scripts#default-values).
+
+### Run a single Node.js script
+
+For many simple, single file projects, you may find it inconvenient to write a
+complete `Dockerfile`. In such cases, you can run a Node.js script by using the
+Node.js Docker image directly:
+
+```console
+$ docker run -it --rm --name my-running-script -v "$PWD":/usr/src/app -w /usr/src/app balenalib/raspberrypi3-node:10 node your-daemon-or-script.js
+```
+
+### Verbosity
+
+Prior to 8.7.0 and 6.11.4 the docker images overrode the default npm log
+level from `warn` to `info`. However due to improvements to npm and new Docker
+patterns (e.g. multi-stage builds) the working group reached a [consensus](https://github.com/nodejs/docker-node/issues/528)
+to revert the log level to npm defaults. If you need more verbose output, please
+use one of the following methods to change the verbosity level.
+
+#### Dockerfile
+
+If you create your own `Dockerfile` which inherits from the `node` image you can
+simply use `ENV` to override `NPM_CONFIG_LOGLEVEL`.
+
+```dockerfile
+FROM balenalib/raspberrypi3-node:10
+ENV NPM_CONFIG_LOGLEVEL info
+...
+```
+
+#### Docker Run
+
+If you run the node image using `docker run` you can use the `-e` flag to
+override `NPM_CONFIG_LOGLEVEL`.
+
+```console
+$ docker run -e NPM_CONFIG_LOGLEVEL=info balenalib/raspberrypi3-node:10 ...
+```
+
+#### NPM run
+
+If you are running npm commands you can use `--loglevel` to control the
+verbosity of the output.
+
+```console
+$ docker run balenalib/raspberrypi3-node:10 npm --loglevel=warn ...
+```
+
+# Image Variants
```

Comment:
Maybe we can have the "Image Variants" section after the "Supported versions" section.

Reply:
Done.